internal/server/singleprocess/state/instance_exec.go

```diff
@@ -60,6 +60,36 @@ type InstanceExec struct {
 	Connected         uint32
 }
 
+func (s *State) InstanceExecCreateByTargetedInstance(exec *InstanceExec) error {
```

Comment:
The `ByDeployment` uses `exec` just as an _out_ param, I think we should just keep following that pattern and pass instance ID as a param here. 

The implementation should set it on `InstanceExec` probably.

Reply:
Sure.